src/main/java/org/veupathdb/service/eda/ds/util/RFileSetProcessor.java

```diff
@@ -18,40 +19,56 @@
     public final String name;
     public final InputStream stream;
     public final Optional<Integer> maxAllowedRows;
+    public final Boolean showMissingness;
+    public final List<String> nonStrataColNames;
     public final ConsumerWithException<RConnection> fileReader;
 
-    public RFileProcessingSpec(String name, InputStream stream, Optional<TwoTuple<Optional<Integer>,BiConsumerWithException<String, RConnection>>> processingInfo) {
+    public RFileProcessingSpec(String name, InputStream stream, Boolean showMissingness, List<String> nonStrataColNames, Optional<TwoTuple<Optional<Integer>,BiConsumerWithException<String, RConnection>>> processingInfo) {
       this.name = name;
       this.stream = stream;
       this.maxAllowedRows = processingInfo.map(i -> i.getFirst()).orElse(Optional.empty());
+      this.showMissingness = showMissingness;
+      this.nonStrataColNames = nonStrataColNames;
       this.fileReader = conn -> processingInfo.map(i -> i.getSecond()).orElse((a,b) -> {}).accept(name, conn);
     }
   }
 
   private final Map<String,InputStream> _dataStreams;
+  private final Map<String, Boolean> _showMissingnessMap = new HashMap<>();
+  private final Map<String, List<String>> _nonStrataColNamesMap = new HashMap<>();
   private final Map<String, TwoTuple<Optional<Integer>,BiConsumerWithException<String, RConnection>>> _processingInfoMap = new HashMap<>();
 
   public RFileSetProcessor(Map<String, InputStream> dataStreams) {
     _dataStreams = dataStreams;
   }
 
   public RFileSetProcessor add(String name, BiConsumerWithException<String, RConnection> fileReader) {
-    return add(name, null, fileReader);
+    return add(name, null, "FALSE", null, fileReader);
   }
 
-  public RFileSetProcessor add(String name, Integer maxAllowedRows, BiConsumerWithException<String, RConnection> fileReader) {
+  public RFileSetProcessor add(String name, Integer maxAllowedRows, String showMissingness, List<String> nonStrataColNames, BiConsumerWithException<String, RConnection> fileReader) {
+    Boolean booleanShowMissingness = showMissingness.equals("FALSE") ? false : true;
```

Comment:
A little more NPE-safe to use:
"FALSE".equals(showMissingness)
which will return false if showMissingness is null rather than throw NPE